sdk/src/Services/DynamoDBv2/Custom/DataModel/Configs.cs

```diff
@@ -81,6 +81,16 @@ public DynamoDBContextConfig()
         /// </summary>
         public bool? SkipVersionCheck { get; set; }
 
+        /// <summary>
+        /// Indicates which DynamoDB table to use. This overrides the table specified 
+        /// by the <see cref="DynamoDBTableAttribute"/> on the .NET objects that you're saving or loading.
+        /// </summary>
+        /// <remarks>
+        /// If you specify this on <see cref="DynamoDBContextConfig"/>, then it will apply to all 
+        /// objects/tables used with that <see cref="DynamoDBContext"/> object unless overriden by an operation-specific config.
+        /// </remarks>
+        public string OverrideTableName { get; set; }
```

Comment:
Having this property on the context's config doesn't make sense to me since the context works with multiple tables. I know you call that out in the doc that it replaces all table mappings. I'm thinking it was a failure in our response to the GitHub issues to not point out that could accomplish what they wanted using the `AWSConfigsDynamoDB.Context.TypeMappings` property. Here is how I configured one of my application to use the table that was created during deployment as part of CloudFormation with CloudFormation naming the table.

```
var tableName = System.Environment.GetEnvironmentVariable(ENV_TABLE_NAME_POLL_DEFINITION);
if (!string.IsNullOrEmpty(tableName))
{
    PollDefinitionTableName = tableName;
    AWSConfigsDynamoDB.Context.TypeMappings[typeof(PollDefinition)] = new Amazon.Util.TypeMapping(typeof(PollDefinition), tableName);
    Console.WriteLine($"Using table {tableName} for PollDefinition");
}
```

If we really want this capability on the context object I think we would have to make it a map of types to table names.

Reply:
I wasn't too attached to this yet either, was mostly throwing it out there since there seems to be interest via the three issues. The only counterargument I can think of is if an application is only working with a single table, the current appraoch is simpler to override.

Should we just point users to `AWSConfigsDynamoDB.Context.TypeMappings`? and then break the hierarchy so it stops looking like you could be able to do this? Or maybe we can leave this PR out for a bit for feedback, I could go comment on the original issues.

Reply:
Yeah I think we should point users to `AWSConfigsDynamoDB.Context.TypeMappings`. You are right that the single property would be useful in a single table design but then confusing in the other case. I don't have a feeling how common single table design is but I don't believe it would be the overwhelming majority to warrant putting in this confusing property for non single table users.

Reply:
You could add a comment on the `OverrideTableName` for the operational config that if users want to make the override global they should use the `AWSConfigsDynamoDB.Context.TypeMappings`.

Reply:
Done in https://github.com/aws/aws-sdk-net/pull/3422/commits/f478a1b33f0b6c2c4c12e525aaf0aaf25318d2c5.

Plus as long as we do the breaking change to separate the hierarchy in V4, if we did want to add a context-level table mapping, that'd be an additive change in the future.